<h1>Palmer Penguin Data pterodactly<h1>
<hr>
<h3>DINOSAHOR Analyst KITSUNE : Stephen C Santos<h3>


In [40]:
import numpy as np
import pandas as pd

In [41]:
penguins = pd.read_csv(r"C:\Users\Admin\Downloads\penguins.csv")
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive Analytics<h1>

In [42]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [43]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Cleaning  Data<h1>

In [44]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [45]:
adelie_df = penguins[penguins['species']=="Adelie"]
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [46]:
gentoo_df = penguins[penguins['species']=="Gentoo"]
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [47]:
chinstrap_df = penguins[penguins['species']=="Chinstrap"]
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [48]:
penguins[penguins["body_mass_g"] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [49]:
penguins[(penguins['species']=="Adelie") & (penguins["body_mass_g"] >= 3000)]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
4         3450.0  female  2007  
5         3650.0    male  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[144 rows x 8 columns]

In [50]:
penguins[(penguins['species']=="Adelie") & (penguins["body_mass_g"] >= 3000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 144 entries, 0 to 151
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            144 non-null    object 
 1   island             144 non-null    object 
 2   bill_length_mm     144 non-null    float64
 3   bill_depth_mm      144 non-null    float64
 4   flipper_length_mm  144 non-null    float64
 5   body_mass_g        144 non-null    float64
 6   sex                140 non-null    object 
 7   year               144 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 10.1+ KB


In [51]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mode()[0])
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mode()[0])
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mode()[0])
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mode()[0])
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [52]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [53]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


<h1> Analytics </h1>

<h3>1. What is the total data entries in each year?</h3>

In [54]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h1>Insights</h1>
<h3>From the given data, 120 was encoded in 2009, 114 data in 2008 and 110 in 2007.</h3>

<H3>2. How many chinstrap data was recorded in year 2008?</H3>

In [55]:
chinstrap_df['year'].value_counts()

year
2007    26
2009    24
2008    18
Name: count, dtype: int64

<h1>Insights</h1>
<h3>The year 2008 in chinstrap data has 24 entries</h3>

<H3>3. How many males are there in gentoo?</H3>

In [56]:
gentoo_df['sex'].value_counts()

sex
male      61
female    58
Name: count, dtype: int64

<h1>Insights</h1>
<h3> Gentoo has 61 males out of 119 entries.</h3>

<H3>4. How many adelie data was recorded in year 2007?</H3>

In [57]:
chinstrap_df['year'].value_counts()

year
2007    26
2009    24
2008    18
Name: count, dtype: int64

<h1>Insights</h1>
<h3>The year 2007 in Adelie data has 26 entries.</h3>

<h3>5. How many adelie is above 2300g? </h3>

In [58]:
adelie_df[adelie_df["body_mass_g"] >= 4500]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7    Adelie  Torgersen            39.2           19.6              195.0   
17   Adelie  Torgersen            42.5           20.7              197.0   
39   Adelie      Dream            39.8           19.1              184.0   
45   Adelie      Dream            39.6           18.8              190.0   
81   Adelie  Torgersen            42.9           17.6              196.0   
101  Adelie     Biscoe            41.0           20.0              203.0   
109  Adelie     Biscoe            43.2           19.0              197.0   
111  Adelie     Biscoe            45.6           20.3              191.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
17        4500.0  male  2007  
39        4650.0  male  2007  
45        4600.0  male  2007  
81        4700.0  male  2008  
101       4725.0  male  2009  
109       4775.0  male  2009  
111       4600.0  male  2009

<h1>Insights</h1>
<h3> Out of Adelie data entries, only 8 of them weigh above 4500g.</h3>

<h3>6. How many gentoo is below 3000g? </h3>

In [59]:
adelie_df[adelie_df["body_mass_g"] <= 3000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
44   Adelie      Dream            37.0           16.9              185.0   
47   Adelie      Dream            37.5           18.9              179.0   
54   Adelie     Biscoe            34.5           18.1              187.0   
58   Adelie     Biscoe            36.5           16.6              181.0   
64   Adelie     Biscoe            36.4           17.1              184.0   
98   Adelie      Dream            33.1           16.1              178.0   
104  Adelie     Biscoe            37.9           18.6              193.0   
116  Adelie  Torgersen            38.6           17.0              188.0   
144  Adelie      Dream            37.3           16.8              192.0   

     body_mass_g     sex  year  
44        3000.0  female  2007  
47        2975.0     NaN  2007  
54        2900.0  female  2008  
58        2850.0  female  2008  
64        2850.0  female  2008  
98        2900.0  female  2008  
104       2925.0  female  2009  
116       2900.0  female  2009  
144       3000.0  female  2009

<h1>Insights</h1>
<h3> Out of the adelie data entries, only 9 of them weigh blow 3000g.</h3>

<h3>7. How many male adelie is there?</h3>

In [60]:
adelie_df[adelie_df["sex"] == 'male']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
5    Adelie  Torgersen            39.3           20.6              190.0   
7    Adelie  Torgersen            39.2           19.6              195.0   
13   Adelie  Torgersen            38.6           21.2              191.0   
14   Adelie  Torgersen            34.6           21.1              198.0   
..      ...        ...             ...            ...                ...   
143  Adelie      Dream            40.7           17.0              190.0   
145  Adelie      Dream            39.0           18.7              185.0   
146  Adelie      Dream            39.2           18.6              190.0   
149  Adelie      Dream            37.8           18.1              193.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g   sex  year  
0         3750.0  male  2007  
5         3650.0  male  2007  
7         4675.0  male  2007  
13        3800.0  male  2007  
14        4400.0  male  2007  
..           ...   ...   ...  
143       3725.0  male  2009  
145       3650.0  male  2009  
146       4250.0  male  2009  
149       3750.0  male  2009  
151       4000.0  male  2009  

[73 rows x 8 columns]

<h1>Insight</h1>
<h3>There's a total of 73 male adelie.</h3>